cli/lsp/language_server.rs

```diff
@@ -774,6 +779,7 @@ impl Inner {
         warn!("Client errored on capabilities.\n{}", err);
       }
     }
+    self.config.update_enabled_paths(self.client.clone()).await;
```

Comment:
This needs to occur after initialized, because the client doesn't actually respond to configuration requests if done earlier.

Reply:
Here it only ever occurs once, when the client has said it has initialised.  This is exactly the point in the lifecycle where you expect to have "blocking" requests, where the server queries the client of anything that falls outside of the initialisation before handling other messages.

I believe there are two other instances, one (on config change) occurs already outside the inner mutex lock (like re-querying the specifiers).  I will double check that the other (when workspace folders are updated) follows that pattern.

Reply:
That's fair.